src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -340,21 +340,47 @@ trait ContextErrors {
 
       //typedSelect
       def NotAMemberError(sel: Tree, qual: Tree, name: Name) = {
-        def errMsg = {
+        import util.{ EditDistance, StringUtil }
+        def errMsg: String = {
           val owner            = qual.tpe.typeSymbol
           val target           = qual.tpe.widen
           def targetKindString = if (owner.isTypeParameterOrSkolem) "type parameter " else ""
           def nameString       = decodeWithKind(name, owner)
           /* Illuminating some common situations and errors a bit further. */
           def addendum         = {
+            val companionSymbol: Symbol = {
```

Comment:
Is this code ever/often called in type errors generated during failed branches of implicit search? If so, we should disable the (expensive) addendum creation. We recently did [something similar](https://github.com/scala/scala/pull/6612/files#diff-05c273d837c49f8b6278b19592a541beR101)

Reply:
I am not sure if it's called in the implicit search.
Inside typedSelect (https://github.com/scala/scala/blob/2b49d718fd2ce034a5f450b2041013e2889591d0/src/compiler/scala/tools/nsc/typechecker/Typers.scala#L4969), `val qual1 = adaptToMemberWithArgs(tree, qual, name, mode)` happens at line 4988, which I am guessing is where implicit search is done. 
At line 5036, typedSelect calls `lookupInQualifier`, which can call `NotAMemberError`. Could that mean that we're ok?

An experiment I did was to add `sys.error(..)` in `alternatives` below, and then added

```scala
object Suggest {
  implicit class RichInt(val i: Int) {
    def flippity: Unit = ()
  }

  1.flippity
}
```

to `pos/implicits-new.scala` test, and it ran without blowing up.

Reply:
Could run all of `partest --pos` and `-partest --run` with that patch? If that goes well, I think we don't need to add any guards.

Reply:
@lrytz Good call. I am seeing a bunch of errors, but they do not look like implicit searches. Most of the failures look like `spec-arrays-pos.log`:

```
error: Int $plus$eq List()
```

likely caused by something like `i += 1`.

Only other kind that I noticed that doesn't involve `$eq` is `t8013-pos.log` etc:

```
error: t8013.Perverse.Impervolator pImpl List()
```

where `pImpl` shows up as a macro implementation:

```scala
def p(args: Any*): String = macro pImpl
```

These occur seldom enough, so I think they are ok, but I'm going to put a guard for `x.endsWith("=")`.